File: src/Utilities/Conversions.cpp
#include "Utilities/Conversions.h"

#include <iostream>

using std::string;
using std::unique_ptr;

namespace Utilities
{

Conversions::Conversions()
{
    initializeValues();
}

void Conversions::initializeValues()
{
}
template <typename T>
void Conversions::printValue(T val)
{
    std::cout<<"going to print value\n";
    std::cout<<val<< "\n";
}

}

File: src/utilities/conversions.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct Conversions {}

impl Conversions {
    pub fn _to_lower_char(val: &mut char) {
        if val.is_alphabetic() {
            *val = val.to_lowercase().next().unwrap();
        }
    }

    pub fn _initialize_values(&self) {}

    pub fn _print_value<T: std::fmt::Debug>(&self, val: T) {
        println!("Going to print value");
        println!("{:?}", val);
    }
}
